### PR TITLE
Update Logging

### PR DESCRIPTION
**Pull request by deepakkinni**
1. Improve logging
2. log the time taken and retry count for DeleteSnapshot.
3. Change from Info to Debug for few log statements.

Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

File: pkg/ivd/ivd_protected_entity.go [hunk 1]
```diff
@@ -291,7 +291,7 @@ func (this IVDProtectedEntity) Snapshot(ctx context.Context, params map[string]m
 	this.logger.Infof("CreateSnapshot called on IVD Protected Entity, %v", this.id.String())
 	var retVal astrolabe.ProtectedEntitySnapshotID
 	retryInterval := time.Second
-	retryCount := 0
+	retryCount := -1
```
Comment (xing-yang) [inline pkg/ivd/ivd_protected_entity.go:294]: What does this mean?  retry indefinitely?

Comment (deepakkinni) [reply pkg/ivd/ivd_protected_entity.go:294]: I want to track only the retries, not the first invocation, hence the -1. If set to 0, in the preceding log msg about retry makes it seem like it was retried once.

**Review by dsu-igeek — APPROVED** [review]
LGTM

File: pkg/ivd/ivd_protected_entity.go [hunk 2]
```diff
@@ -392,8 +393,11 @@ func (this IVDProtectedEntity) DeleteSnapshot(ctx context.Context, snapshotToDel
 			}
 			return false, errors.Wrapf(err, "Failed to create a task for the DeleteSnapshot invocation on IVD Protected Entity, %v, with input arg, %v", this.GetID().String(), snapshotToDelete.String())
 		}
-
+		this.logger.Infof("Retrieved VSLM task %s to track DeleteSnapshot on IVD %s", vslmTask.Value, this.id.String())
```
Comment (xing-yang) [inline pkg/ivd/ivd_protected_entity.go:396]: this.id.String() -> this.GetID().String()

File: pkg/ivd/ivd_protected_entity.go [hunk 3]
```diff
@@ -444,6 +448,7 @@ func (this IVDProtectedEntity) GetID() astrolabe.ProtectedEntityID {
 
 func (this IVDProtectedEntity) Overwrite(ctx context.Context, sourcePE astrolabe.ProtectedEntity, params map[string]map[string]interface{},
 	overwriteComponents bool) error {
+	this.logger.Infof("Overwriting current IVD PE %s with the source PE %s", this.id.String(), sourcePE.GetID().String())
```
Comment (xing-yang) [inline pkg/ivd/ivd_protected_entity.go:451]: this.GetID().String()

File: pkg/ivd/ivd_protected_entity.go [hunk 4]
```diff
@@ -467,6 +472,7 @@ func (this IVDProtectedEntity) Overwrite(ctx context.Context, sourcePE astrolabe
 	if err != nil {
 		return errors.Wrapf(err, "Data copy failed from PE %s, to PE %s", this.GetID().String(), sourcePE.GetID().String())
 	}
+	this.logger.Infof("Successfully Overwrote current IVD PE %s with the source PE %s", this.id.String(), sourcePE.GetID().String())
```
Comment (xing-yang) [inline pkg/ivd/ivd_protected_entity.go:475]: this.GetID().String()

Comment (deepakkinni) [reply pkg/ivd/ivd_protected_entity.go:475]: done.

Comment (xing-yang) [reply pkg/ivd/ivd_protected_entity.go:475]: Looks like this is not changed?

Comment (deepakkinni) [reply pkg/ivd/ivd_protected_entity.go]: fixed it now.

**Review by lintongj — APPROVED** [review]
/lgtm

**Review by xing-yang — APPROVED** [review]
LGTM

**Pull request merged**